File: app/src/main/java/com/alex/morgan/bearlist/list/BearListModule.java
package com.alex.morgan.bearlist.list;

import java.util.Random;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
abstract class BearListModule {

    @Binds
    abstract BearListContract.Presenter bindPresenter(BearListPresenter presenter);

    @Binds
    abstract BearSource bindBearGenerator(RandomBearGenerator bearGenerator);

    @Provides
    static Random provideRandom() {
        return new Random();
    }
}

File: app/src/main/java/com/alex/morgan/bearlist/list/BearListModule.kt
package com.alex.morgan.bearlist.list

import com.alex.morgan.FragmentScope
import dagger.Binds
import dagger.Module

@Module(includes = [BearDataModule::class])
internal abstract class BearListModule {
    @Binds
    @FragmentScope
    abstract fun bindBearListPresenter(presenter: BearListPresenter): BearListContract.Presenter
}
